card/src/main/java/com/adyen/checkout/card/repository/DefaultAddressRepository.kt

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2022 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by ozgur on 5/8/2022.
+ */
+
+package com.adyen.checkout.card.repository
+
+import android.util.LruCache
+import com.adyen.checkout.card.api.AddressService
+import com.adyen.checkout.card.api.model.AddressItem
+import com.adyen.checkout.card.ui.AddressFormInput
+import com.adyen.checkout.components.base.Configuration
+import com.adyen.checkout.core.api.Environment
+import com.adyen.checkout.core.log.LogUtil
+import com.adyen.checkout.core.log.Logger
+import com.adyen.checkout.core.util.runSuspendCatching
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.launch
+import java.util.Locale
+
+internal class DefaultAddressRepository : AddressRepository {
+
+    private val _statesFlow: MutableSharedFlow<List<AddressItem>> = MutableSharedFlow(0, 1, BufferOverflow.DROP_OLDEST)
+    override val statesFlow: Flow<List<AddressItem>> = _statesFlow
+
+    private val _countriesFlow: MutableSharedFlow<List<AddressItem>> =
+        MutableSharedFlow(0, 1, BufferOverflow.DROP_OLDEST)
+    override val countriesFlow: Flow<List<AddressItem>> = _countriesFlow
+
+    private val cache: LruCache<String, List<AddressItem>> = LruCache<String, List<AddressItem>>(CACHE_ENTRY_SIZE)
```

Comment:
Isn't this cache able to basically able to cache every possible country? Because that would defeat the purpose of `LruCache`. Also, since there are not a lot of entries in this cache I think using a simple `HashMap` is more efficient